recipes/newrelic/apm/php-agent/php.yml

```diff
@@ -67,3 +70,47 @@ install:
           else
             echo "You will need to restart your PHP application in order for instrumentation to be enabled."
           fi
+
+    send_transaction:
+      cmds:
+        - echo "Send queryable transactions."
+        - |
+          TMP_INSTALL_DIR="/tmp/php-agent-install"
+          cd $TMP_INSTALL_DIR
+          #
+          # Expand the log files from the most recent installation attempt,
+          # and get the name of the most recent log file.
+          #
+          tar -xvf $(ls -t nrinstall*.tar | head -1)
+          NR_INSTALL_LOG=$(ls -t nrinstall*.log | head -1)
+          #
+          # Get the web server PHP INI directories that the most recent 
+          # installation attempt used.
+          #
+          WEB_PHP_INI_DIR=$(sed -n 's/.*pi_inidir_dso=\(.*\)\/conf.d/\1/p' $NR_INSTALL_LOG)
+          WEB_NR_INI_DIR=$(sed -n 's/.*pi_inidir_dso=\(.*\/\)/\1/p' $NR_INSTALL_LOG)
+          if [ -z "${WEB_PHP_INI_DIR}" ]; then
+              WEB_PHP_INI_DIR=$WEB_NR_INI_DIR
+          fi
+          #
+          # The first transaction initializes the app.
+          # The second transaction is for luck.
+          # The third time’s the charm.
+          #
+          if [ -n "${WEB_PHP_INI_DIR}" ]; then
+          #
+          # Web server was detected. Send transaction associated with the webserver php.ini
+          # and the newrelic.ini file associated with that web server.
+          #
+             php -c "${WEB_PHP_INI_DIR}/php.ini" -c "${WEB_NR_INI_DIR}/newrelic.ini" -n --ini &>/dev/null
+             php -c "${WEB_PHP_INI_DIR}/php.ini" -c "${WEB_NR_INI_DIR}/newrelic.ini" -n --ini &>/dev/null
+             php -c "${WEB_PHP_INI_DIR}/php.ini" -c "${WEB_NR_INI_DIR}/newrelic.ini" -n --ini &>/dev/null
+          else
+          #
+          # No web server detected, this was a pure PHP CLI php agent installation.
+          #
+             php —-ini
+             php —-ini
+             php —-ini
```

Comment:
I expect we will want to know how many of our virtuoso based installs go through this path vs the web path. How do other Virtuoso based agent installs track such things? A supportability metric is what comes to mind, but I don't think we have an exposed API to send one. Creating a custom PHP script for each path here would put the data in the customer's account which doesn't help us look for trends across the install base. Are we only able to track failure causes via return codes currently?

This might be something we need to sort out later. I just wanted to start the conversation.